code.js

```diff
@@ -1,3 +1,26 @@
 function tsp_hk(distance_matrix) {
-    return -1;
+    if (distance_matrix.length <= 1) return 0
+    min_val = Infinity
+    cities = []
+    for (var i = 0; i < distance_matrix.length; i++) {cities.push(i)}
+    for (var i = 0; i < distance_matrix.length; i++) {
+        min_val = Math.min(min_val, tspPath(distance_matrix, i,cities,[]))
+    }
+    return min_val
+}
+function tspPath(graph, start, cities,memory){
+    place = JSON.stringify(cities)
+    if(memory[place] != null){return memory[place]}
```

Comment:
Don't you also have to take the start city into account? The result might be different for the same set of cities, but different start cities.